lisa/util/shell.py

```diff
@@ -31,6 +51,638 @@
+
+                success = True
+
+            except (socket.error, EOFError) as err:
+                last_err = err
+
+                if self._sftp is not None:
+                    self._sftp.close()
+                    self._sftp = None
+
+                time.sleep(self.retry_period)
+
+        if not success:
+            raise ConnectionError(
+                "Failed to execute an SFTP command after {} retries"
+                " due to connection failure: {}".format(
+                    self.max_retries, last_err))
+
+        return method(self._sftp) # type: ignore
+
+    def chmod(self, path: str, mode: int):
+        """See paramiko.SFTP documentation."""
+        return self.__wrap(method=lambda sftp: sftp.chmod(path, mode))
+    
+    def chown(self, path: str, uid: int, gid: int):
+        """See paramiko.SFTP documentation."""
+        return self.__wrap(method=lambda sftp: sftp.chown(path, uid, gid))
+
+    def stat(self, path: str):
+        """See paramiko.SFTP documentation."""
+        return self.__wrap(method=lambda sftp: sftp.stat(path))
+    
+    def rmdir(self, path: str):
+        """See paramiko.SFTP documentation."""
+        return self.__wrap(method=lambda sftp: sftp.rmdir(path))
+
+    def listdir_attr(self, path: str = '.') -> List[paramiko.SFTPAttributes]:
+        """See paramiko.SFTP documentation."""
+        return self.__wrap(method=lambda sftp: sftp.listdir_attr(path))
+
+    def remove(self, path: str) -> None:
+        """See paramiko.SFTP documentation."""
+        return self.__wrap(method=lambda sftp: sftp.remove(path))
+
+    unlink = remove
+
+    def posix_rename(self, oldpath: str, newpath: str):
+        """See paramiko.SFTP documentation."""
+        return self.__wrap(method=lambda sftp: sftp.posix_rename(oldpath, newpath))
+
+    def mkdir(self, path: str, mode: int = 0o777):
+        """See paramiko.SFTP documentation."""
+        return self.__wrap(method=lambda sftp: sftp.mkdir(path, mode))
+
+    def lstat(self, path: str):
+        """See paramiko.SFTP documentation."""
+        return self.__wrap(method=lambda sftp: sftp.lstat(path))
+
+    def symlink(self, source: str, dest: str):
+        """See paramiko.SFTP documentation."""
+        return self.__wrap(method=lambda sftp: sftp.symlink(source, dest))
+
+    def put(
+        self,
+        localpath: str,
+        remotepath: str,
+        callback: Optional[Callable[[int, int], Any]] = None,
+        confirm: bool = True,
+    ):
+        """See paramiko.SFTP documentation."""
+        return self.__wrap(method=lambda sftp: sftp.put(localpath, remotepath, callback, confirm))
+
+    def get(self, remotepath: str, localpath: str, callback: Optional[Callable[[int, int], Any]] = None):
+        """See paramiko.SFTP documentation."""
+        return self.__wrap(method=lambda sftp: sftp.get(remotepath, localpath, callback))
+
+class SshShellWithSFTP(icontract.DBC):
```

Comment:
What this is used to fix? Can we just include the core parts?

Reply:
yes I'm trying to only include the functions being used in lisa, I don't want to break current lisa functions or introduce unused contents.

BTW you can ignore this draft for now I'm trying to fix some linting alerts